lib/sidekiq/instrument/middleware/server.rb

```diff
@@ -10,16 +10,17 @@ class ServerMiddleware
     def call(worker, job, queue, &block)
       Statter.statsd.increment(metric_name(worker, 'dequeue'))
       Statter.dogstatsd&.increment('sidekiq.dequeue', worker_dog_options(worker))
+      WorkerMetrics.trace_workers_decrement_counter(worker.class.to_s.underscore)
 
       start_time = Time.now
-      WorkerMetrics.trace_workers_decrement_counter(worker.class.to_s.underscore)
       yield block
       execution_time_ms = (Time.now - start_time) * 1000
       Statter.statsd.measure(metric_name(worker, 'runtime'), execution_time_ms)
       Statter.dogstatsd&.timing('sidekiq.runtime', execution_time_ms, worker_dog_options(worker))
     rescue StandardError => e
       Statter.statsd.increment(metric_name(worker, 'error'))
       Statter.dogstatsd&.increment('sidekiq.error', worker_dog_options(worker))
+      WorkerMetrics.trace_workers_decrement_counter(worker.class.to_s.underscore)
```

Comment:
This means if you rescue an error raised on line 16 for example you'll end up decrementing your counter twice. Is that really what you want?

Reply:
Feels like what you're really looking for is an `ensure`